src/components/genericEmbed.tsx

```diff
@@ -0,0 +1,45 @@
+// ----- Imports ----- //
+
+import { css } from '@emotion/core';
+import { remSpace } from '@guardian/src-foundations';
+import { text } from '@guardian/src-foundations/palette';
+import { textSans } from '@guardian/src-foundations/typography';
+import { withDefault } from '@guardian/types';
+import type { Generic } from 'embed';
+import { maybeRender } from 'lib';
+import type { FC } from 'react';
+import React from 'react';
+import { darkModeCss } from 'styles';
+
+// ----- Component ----- //
+
+const styles = css`
+	margin: ${remSpace[4]} 0;
+
+	${darkModeCss`
+        background: white;
+        padding: ${remSpace[2]};
+    `}
+`;
+
+const captionStyles = css`
+	${textSans.xsmall()}
+	color: ${text.supporting};
+`;
+
+interface Props {
+	embed: Generic;
+}
+
+const GenericEmbed: FC<Props> = ({ embed }) => (
+	<figure css={styles}>
+		<iframe srcDoc={embed.html} title={withDefault('Embed')(embed.alt)} />
```

Comment:
I realise this is WIP but in this case the iframe just takes on a default size that does not match the size of the embedded content.

For example:

http://localhost:8080/culture/2019/apr/27/idris-elba-marries-sabrina-dhowre-in-morocco

That is an instagram embed in an 'embed' type element.

Again as Im sure you know, the instagram js also gets blocked by the CSP.

Reply:
I've done some work to handle the height issue in 42a6e61. This only works for some generic embeds (the ones that are already wrapped in an `iframe` with `height`s).

Generic instagram embeds having their JS blocked is already an issue on `main` - I'll look at it in a separate PR 👍.